c7n/resources/workspaces.py

```diff
@@ -91,3 +93,51 @@ def get_resource_value(self, k, i):
 class KmsFilter(KmsRelatedFilter):
 
     RelatedIdsExpression = 'VolumeEncryptionKey'
+
+
+@resources.register('workspaces-image')
+class WorkspaceImage(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'workspaces'
+        enum_spec = ('describe_workspace_images', 'Images', None)
+        arn_type = 'workspaceimage'
+        name = id = 'ImageId'
+        universal_taggable = True
+
+    augment = universal_augment
+
+
+@WorkspaceImage.filter_registry.register('cross-account')
+class WorkspaceImageCrossAccount(CrossAccountAccessFilter):
+
+    schema = type_schema(
+        'cross-account',
+        # white list accounts
+        whitelist_from=ValuesFrom.schema,
+        whitelist={'type': 'array', 'items': {'type': 'string'}})
+
+    permissions = ('workspaces:DescribeWorkspaceImagePermissions',)
+
+    def process(self, resources, event=None):
```

Comment:
Rather than overriding `process()` here, it may be cleaner to override `get_resource_policy()` to get the image permissions. `CrossAccountAccessFilter` can handle it from there, since it already has logic to compare against allowed accounts and add an annotation for any violations. #5622 should be a useful reference for how that refactor could work.

Reply:
Hi AJ. Would this approach work for the response given by describe_workspace_image_permissions since its a list of account ids, not a policy document?

Reply:
Oh that was silly, you're right! Nevermind, carry on doing things that work :).